core/domain/user_jobs_one_off.py

```diff
@@ -95,27 +101,42 @@ def entity_classes_to_map_over(cls):
     def map(item):
         """Implements the map function for this job."""
 
-        # Assumption: We do not have any profile user in the UserSettingsModel,
-        # because we have no method of associating profile users with a full
-        # user.
+        # Assumptions:
+        # 1) We are only having full users in the database, and no
+        #    profile users. For profiles we need to do a role check and ensure
+        #    the gae_id attribute is None for them.
+        # 2) This migration works correctly only if gae_id cannot be updated
+        #   in any way for a user. If a gae_id is mapped to some other user by
+        #   deleing it, an error will be raised while migration (see below).
 
         user_auth_model = user_models.UserAuthModel.get_by_auth_id(
             feconf.AUTH_METHOD_GAE, item.gae_id)
-        if (user_auth_model is not None) and (user_auth_model.id != item.id):
-            yield ('GAE_ID %s already registered with user_id %s.' % (
-                item.gae_id, user_auth_model.id), 1
-                  )
+        if user_auth_model is not None:
+            if user_auth_model.id != item.id:
+                yield (
+                    'Error, gae_id already existing',
+                    {'user_id': item.id, 'gae_id': item.gae_id}
+                )
+            else:
+                if (item.deleted is True and
+                        user_auth_model.deleted is not True):
```

Comment:
```suggestion
                if item.deleted and not user_auth_model.deleted:
```

Reply:
Not required now. Please see the updated migration job.